Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/non_rigid_mesh_registration.h

```diff
@@ -0,0 +1,930 @@
+
+  typedef typename CGAL::internal_np::Lookup_named_param_def<internal_np::correspondences_t, NamedParameters1, std::vector<std::pair<typename boost::graph_traits<TriangleMesh1>::vertex_descriptor, typename boost::graph_traits<TriangleMesh2>::vertex_descriptor>>>::reference Correspondences_type;
+  Correspondences_type correspondences = CGAL::parameters::choose_parameter(CGAL::parameters::get_parameter_reference(np1, CGAL::internal_np::correspondences), std::vector<std::pair<typename boost::graph_traits<TriangleMesh1>::vertex_descriptor, typename boost::graph_traits<TriangleMesh2>::vertex_descriptor>>());
+
+  using Gt1 = typename GetGeomTraits<TriangleMesh1, NamedParameters1>::type;
+
+  typedef std::pair<typename Gt1::Point_3, typename Gt1::Vector_3> Point_with_normal;
+  typedef std::vector<Point_with_normal> Pwn_vector;
+  typedef CGAL::First_of_pair_property_map<Point_with_normal>  Point_map;
+  typedef CGAL::Second_of_pair_property_map<Point_with_normal> Normal_map;
+  Pwn_vector points;
+  points.reserve(target.num_vertices());
+
+  for (auto v : target.vertices())
+    points.push_back(std::make_pair(get(vpm, v), get(vnm, v)));
+
+  std::vector<std::pair<typename boost::graph_traits<TriangleMesh1>::vertex_descriptor, std::size_t>> correspondences_pts;
+  correspondences_pts.reserve(correspondences.size());
+  for (auto p : correspondences)
+    correspondences_pts.push_back(std::make_pair(p.first, static_cast<std::size_t>(p.second)));
+
+  non_rigid_mesh_to_points_registration(source, points, vtm, vrm, np1.combine(parameters::correspondences(correspondences_pts)), parameters::point_map(Point_map()).normal_map(Normal_map()));
+
+#else
+  static_assert(false, "Eigen library is required for non-rigid mesh registration");
+#endif
+}
+
+/*!
+* \ingroup PMP_registration_grp
+*
+* \brief applies a non-rigid transformation to the vertices of the mesh. Face and vertex normal vectors are invalid after transformation.
+*
+* @tparam TriangleMesh a model of `MutableFaceGraph`.
+* @tparam VertexTranslationMap is a property map with `boost::graph_traits<TriangleMesh1>::%vertex_descriptor`
+ *   as key type and a \cgal Kernel `Vector_3` as value type.
+* @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters".
+*
+* @param mesh the triangle mesh to be transformed.
+* @param vtm a readable vertex property map of `mesh` to store the translation vector of the registration.
+* @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below.
+*
+* \cgalNamedParamsBegin
+*   \cgalParamNBegin{geom_traits}
+*     \cgalParamDescription{an instance of a geometric traits class}
+*     \cgalParamType{a class model of `Kernel`}
+*     \cgalParamDefault{a \cgal Kernel deduced from the point type, using `CGAL::Kernel_traits`}
+*     \cgalParamExtra{The geometric traits class must be compatible with the vertex point type.}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{vertex_point_map}
```

Comment:
don't you also need the vertex normal map for applying the rotation matrix?

Reply:
I did not manage to handle an optional vertex normal map.